core/lib/vm_interface/src/vm.rs

```diff
@@ -44,9 +44,6 @@ pub trait VmInterface {
         with_compression: bool,
     ) -> (BytecodeCompressionResult<'_>, VmExecutionResultAndLogs);
 
-    /// Record VM memory metrics.
-    fn record_vm_memory_metrics(&self) -> VmMemoryMetrics;
```

Comment:
AFAIU, metrics are specific to the old tracer-powered VM implementation, and these metrics are not critical for correctness (they are only used for metrics). I haven't found a way to fill them for the new VM.